File: app/src/main/java/com/katkov/lolachievements/application/ui/main/MainPresenter.java
package com.katkov.lolachievements.application.ui.main;

import com.arellomobile.mvp.InjectViewState;
import com.arellomobile.mvp.MvpPresenter;
import com.katkov.lolachievements.application.navigation.Screens;
import com.katkov.lolachievements.domain.model.EntryInfoModel;
import com.katkov.lolachievements.data.local.prefser.EntryInfoHolder;

import javax.inject.Inject;

import ru.terrakok.cicerone.Router;
import ru.terrakok.cicerone.Screen;

@InjectViewState
public class MainPresenter extends MvpPresenter<MainView> {

    private final Router router;
    private final EntryInfoHolder entryInfoHolder;

    @Inject
    MainPresenter(Router router, EntryInfoHolder entryInfoHolder) {
        this.router = router;
        this.entryInfoHolder = entryInfoHolder;
    }

    @Override
    protected void onFirstViewAttach() {
        super.onFirstViewAttach();
        checkAlreadyLogged();
    }

    private void checkAlreadyLogged() {
        EntryInfoModel entryInfoModel = entryInfoHolder.getEntryInfoModel();
        router.newRootScreen(getFirstScreen(entryInfoModel));
    }

    private Screen getFirstScreen(EntryInfoModel entryInfoModel) {
        if (entryInfoModel == null) {
            return new Screens.FirstEntryScreen();
        } else {
            return new Screens.CheckFirstEntryInfoScreen();
        }
    }
}

File: app/src/main/java/com/katkov/lolachievements/application/ui/main/MainPresenter.kt
package com.katkov.lolachievements.application.ui.main

import com.arellomobile.mvp.InjectViewState
import com.katkov.lolachievements.application.base.BasePresenter
import com.katkov.lolachievements.application.navigation.Screens
import com.katkov.lolachievements.di.annotations.GlobalRouter
import com.katkov.lolachievements.domain.interactor.LoginInteractor
import com.katkov.lolachievements.domain.model.LoginModel
import ru.terrakok.cicerone.Router
import ru.terrakok.cicerone.Screen
import javax.inject.Inject

@InjectViewState
class MainPresenter
@Inject
internal constructor(
    @GlobalRouter val router: Router,
    private val loginInteractor: LoginInteractor
) : BasePresenter<MainView>() {

    override fun onFirstViewAttach() {
        super.onFirstViewAttach()
        checkAlreadyLogged()
    }

    private fun checkAlreadyLogged() {
        val loginModel = loginInteractor.getLoginModel()
        router.newRootScreen(getFirstScreen(loginModel))
    }

    private fun getFirstScreen(loginModel: LoginModel?): Screen {
        return if (loginModel == null) {
            Screens.LoginScreen()
        } else {
            Screens.BottomNavigationFragmentScreen()
        }
    }
}
